pkg/network/dhcp/server/server.go

```diff
@@ -100,6 +108,35 @@ func closeDHCPServerIgnoringError(l ServeIfConn) {
 	}
 }
 
+func removedhcpStartedFile(dhcpStartedFile string) {
+	err := os.Remove(dhcpStartedFile)
+	if err != nil {
+		log.Log.Warningf("failed to remove dhcpStartedFile %s: %v", dhcpStartedFile, err)
+	}
+}
+
+func stopOnIfaceDeletion(serverIface string, l ServeIfConn) {
```

Comment:
Could you please share why the method has been chosen to terminate the DHCP server?
Specifically, why not end the server using a context from the upper stack when the unplug is detected?

The current method and its addition lacks unit tests per what I can see.
Testing go-routines is challenging, do you think it can be done in this case?